### DISCUSSION TITLE
Why is the text color of the Typography component black?

### DISCUSSION THREAD

**Discussion by sawa-ko** [ant-design/ant-design#48851] [Q&A]
Actually I am facing the problem that I have configured the dark theme algorithm in my config provider, so I want to use the typography components, but they have black text color and not white as it should be.

No matter where you use it, it is always black.

| ![image](https://github.com/ant-design/ant-design/assets/56084970/b9b02f6e-24ed-4c0e-abaf-729689179ad9)| ![image](https://github.com/ant-design/ant-design/assets/56084970/f95404f7-b7ec-4c7c-8d18-e818022f9e81)|
|--------|--------|

```tsx
    <>
      <Title>Text</Title>
      <Card>
        <Title>Card</Title>
      </Card>
    </>
```

```tsx
const { darkAlgorithm, defaultAlgorithm } = theme;

export default function Providers({ children, i18n }: ProvidersProps) {
  return (
    <NextIntlClientProvider locale={i18n.locale} messages={i18n.messages} timeZone={LOCALE_TIMEZONE}>
      <AntdRegistry>
        <ConfigProvider
          theme={{
            algorithm: darkAlgorithm,
            cssVar: true,
            hashed: false,
          }}
        >
          {children}
        </ConfigProvider>
      </AntdRegistry>
    </NextIntlClientProvider>
  );
}
```

**Comment by echoboomer** [comment]
Would love an answer to this. Was considering picking up this framework, but seeing a similar issue when attempting to use components in a TypeScript project using a lot of Chakra under the hood already, including theme management. No matter what, font is always black when importing Ant Design components.

**Comment by AsuraKev** [comment]
maybe try set a differernt color through config provider for dark mode?

**Comment by Wxh16144** [comment]
It's normal on my end, I'm using the latest Chrome browser and antd.

```tsx
import React from 'react'
import { ConfigProvider, theme, Typography, Card, version } from 'antd'

function App() {
  return (
    <div
      style={{
        height: '100vh',
        width: '100vw',
        background: 'gray'
      }}
    >
      <ConfigProvider
        theme={{
          algorithm: theme.darkAlgorithm,
          cssVar: true,
          hashed: false,

        }}
      >
        <Typography.Title level={1}>Database</Typography.Title>
        <Card
          title="Database"
        >
          <p>Database</p>
          <h4>antd v:{version}</h4>
          <Typography.Title level={1}>Database</Typography.Title>
        </Card>
      </ConfigProvider>
    </div>
  )
}

export default App
```

<img width="937" alt="image" src="https://github.com/user-attachments/assets/5b0bb6e7-4e6e-454b-bce8-4ce44b85c091">

**Reply by echoboomer** [reply]
@Wxh16144 You are correct, that was my mistake. I added `ConfigProvider` to root and used a ternary to set `algorithm` to light or dark based on the session storage value that Chakra sets for whether the theme is light or dark. It does require a reload to actually change the objects rendered by `antd`, but it works. I'm not sure there's a better way since theming is handled by Chakra in my case.
